frontend/src/components/App/Settings/SettingsCluster.tsx

```diff
@@ -314,6 +314,24 @@ export default function SettingsCluster() {
     );
   }
 
+  // Display the original name of the cluster if it was loaded from a kubeconfig file.
+  function ClusterName() {
+    const originalName = clusterInfo?.meta_data?.originalName;
+    const source = clusterInfo?.meta_data?.source;
+    const isNonDynamic = source === 'kubeconfig';
+
+    return (
+      <>
+        <Typography>{t('translation|Name')}</Typography>
+        {isNonDynamic && (
```

Comment:
and then this could be replaced with
```suggestion
        {displayOriginalName && (
```

Reply:
nice!